featurebyte/models/target.py

```diff
@@ -0,0 +1,73 @@
+"""
+This module contains Target related models
+"""
+from __future__ import annotations
+
+from typing import List
+
+import pymongo
+from pydantic import Field, validator
+
+from featurebyte.common.validator import duration_string_validator
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueConstraintResolutionSignature,
+    UniqueValuesConstraint,
+)
+from featurebyte.query_graph.graph import QueryGraph
+
+
+class TargetModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    Model for Target
+
+    id: PydanticObjectId
+        Target id of the object
+    name: str
+        Name of the Target
+    created_at: datetime
+        Datetime when the Target object was first saved or published
+    updated_at: datetime
+        Datetime when the Target object was last updated
+    """
+
+    # Recipe
+    graph: QueryGraph = Field(allow_mutation=False)
+    node_name: str
+
+    # These fields will either be inferred from the recipe, or manually provided by the user only if they're creating
+    # a target without a recipe.
+    window: str
```

Comment:
Does this refer to the "horizon" field in the spec?

Reply:
yea you're right! i'll fix this in a follow-up PR